buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java

```diff
@@ -382,6 +385,8 @@ protected ByteBuf newHeapBuffer(int initialCapacity, int maxCapacity) {
                     new UnpooledHeapByteBuf(this, initialCapacity, maxCapacity);
         }
 
+        incrementUsedHeapBytes(initialCapacity);
```

Comment:
I think we should use the size of the actual memory consumed by the buffer, rather than the sum of the capacities of all in-use buffers. So these increments and decrements will be in terms of pool chunk sub-page element size (for "small" allocation), pool chunk page-multiple run size (for "normal" allocations), or the requested capacity (for "huge" allocations).

Reply:
In other words, I'd like to see this accounting be pushed down to where memory is removed from, or returned to, the pool.